doc/sourmash-internals.md

```diff
@@ -0,0 +1,648 @@
+
+### Making sketches
+
+Sketches are produced by hashing k-mers with murmurhash and then
+keeping either the lowest `num` hashes (for MinHashes sketches) or
+keeping all hashes below `2**64 / scaled` (for FracMinHash sketches).
+This has the effect of selecting approximately one hash for every
+`scaled` k-mers - so, when sketching a set of 100,000 distinct k-mers,
+a scaled value of 1,000 would yield approximately 100 hashes to be
+retained in the sketch.
+
+The default MinHash sketches use parameters so that they are
+compatible with mash sketches.
+
+See [utils/compute-dna-mh-another-way.py](https://github.com/sourmash-bio/sourmash/blob/latest/utils/compute-dna-mh-another-way.py) for details on how k-mers are
+hashed.
+
+Note that if hashes are produced some other way (with a different hash
+function) or from some source other than DNA, sourmash can still work
+with them; only `sourmash sketch` does input checking.
+
+### Compatibility checking
+
+The point of the signatures and sketches is to enable certain kinds of
+rapid comparisons - Jaccard similarity and overlap analysis,
+specifically. However, these comparisons can only be done between
+compatible sketches.
+
+Here, "compatible" means -
+* the same MurmurHash seed (default 42);
+* the same k-mer size/ksize (see k-mer sizes, below);
+* the same molecule type (see molecule types, below);
+* the same `num` or `scaled` (although see [this downsampling discussion](api-example.md#downsampling-signatures), and the next two sections);
+
+sourmash uses selectors (`Index.select(...)`) to select sketches with
+compatible ksizes, molecule types, and sketch types.
+
+### Scaled (FracMinHash) sketches support similarity and containment
+
+Per our discussion in [Irber et al., 2022](https://www.biorxiv.org/content/10.1101/2022.01.11.475838v2), FracMinHash sketches can always be compared
+by downsampling to the max of the two scaled values.  (This is not always
+true of sketches stored in indexed collections, e.g. SBTs; see [sourmash#1799](https://github.com/sourmash-bio/sourmash/issues/1799).)
+
+In practice, sourmash does all necessary downsampling dynamically, but
+returns the original sketches. This means that (for example) you can
+do a low-resolution/high-scaled large scale search by specifying a
+high scaled value, and then do a higher resolution comparison with
+only the highly similar matches the results to do a more refined (see
+below, Speeding up `gather` and `search`.)
+
+### Num (MinHash) sketches support Jaccard similarity
+
+"Regular" MinHash (or "num MinHash") sketches are implemented the same
+way as in mash.  However, they are less well supported in sourmash,
+because they don't offer the same opportunities for metagenome
+analysis.  (See also [sourmash#1354](https://github.com/sourmash-bio/sourmash/issues/1354).)
+
+Num MinHash sketches can always be compared by downsampling to a
+common `num` value. This may need to be done manually using `sourmash
+sig downsample`, however.
+
```

Comment:
```suggestion

### Conversion between Scaled (FracMinHash) and Num (MinHash) signatures with `downsample`

As discussed in the previous sections, it is possible to adjust the `scaled` and `num` values to compare two FracMinHash signatures or two Num MinHash signatures. However, it is also possible to covert between the `scaled` and `num` signatures with the `sourmash sig downsample` command. For more details, review [command line](https://sourmash.readthedocs.io/en/latest/command-line.html#sourmash-signature-downsample-decrease-the-size-of-a-signature) document.
```
I think this would be a good spot for a section on scaled and num conversions because the two previous sections covered the conversion within the scaled and num domain, but the explicit discussion of converting between them is missing.